app/cli/internal/policydevel/lint.go

```diff
@@ -0,0 +1,390 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policy
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/open-policy-agent/opa/v1/format"
+	"github.com/styrainc/regal/pkg/linter"
+	"github.com/styrainc/regal/pkg/rules"
+	"gopkg.in/yaml.v3"
+)
+
+type Policy struct {
+	Path      string
+	YAMLFiles []*File
+	RegoFiles []*File
+	Format    bool
+}
+
+type File struct {
+	Path    string
+	Content []byte
+}
+
+type EmbeddedPolicy struct {
+	Kind string
+	Rego string
+}
+
+// Read policy files from the given directory
+func Read(absPath string, format bool) (*Policy, error) {
+	fileInfo, err := os.Stat(absPath)
+	if os.IsNotExist(err) {
+		return nil, fmt.Errorf("path does not exist: %s", absPath)
+	}
+
+	policy := &Policy{
+		Path:   absPath,
+		Format: format,
+	}
+
+	if fileInfo.IsDir() {
+		// Read all *.yaml and *.rego files in the directory
+		files, err := os.ReadDir(absPath)
+		if err != nil {
+			return nil, fmt.Errorf("reading directory: %w", err)
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+
+			filePath := filepath.Join(absPath, file.Name())
+			content, err := os.ReadFile(filePath)
+			if err != nil {
+				return nil, fmt.Errorf("reading %s: %w", file.Name(), err)
+			}
+
+			switch strings.ToLower(filepath.Ext(file.Name())) {
+			case ".yaml", ".yml":
+				policy.YAMLFiles = append(policy.YAMLFiles, &File{
+					Path:    filePath,
+					Content: content,
+				})
+			case ".rego":
+				policy.RegoFiles = append(policy.RegoFiles, &File{
+					Path:    filePath,
+					Content: content,
+				})
+			}
+
+			if len(policy.YAMLFiles) == 0 && len(policy.RegoFiles) == 0 {
+				return nil, fmt.Errorf("directory must contain at least one .yaml/.yml or .rego file")
+			}
+		}
+	} else {
+		// Read given file
+		content, err := os.ReadFile(absPath)
+		if err != nil {
+			return nil, fmt.Errorf("reading file: %w", err)
+		}
+
+		switch strings.ToLower(filepath.Ext(absPath)) {
```

Comment:
this code seems repeated ot the one above. Maybe this detection can be extracted? It looks like a good opportunity to maybe to a recursive function when you find potential candidate files in a directory :)